### ISSUE TITLE
Area2D::_clear_monitoring error after queue_free()

### ISSUE THREAD

**Issue by fredlim** [godotengine/godot#1247]
After sync to github I got errors in console:

This function cause the area_2d error

```
func _on_Area2D_body_enter( body ):
    if ( body.is_in_group("bullet") ):
        body.queue_free()       <----------------------- this line cause error
        get_node("../..").queue_free()
```

```
ERROR: Memory::free_static:  Condition ' !MemoryPoolStatic::get_singleton() ' is
 true.
At:  core\os\memory.cpp:58
ERROR: Memory::free_static:  Condition ' !MemoryPoolStatic::get_singleton() ' is
 true.
At:  core\os\memory.cpp:58
ERROR: DirAccess::make_dir_recursive:  Method/Function Failed, returning: err
At:  core\os\dir_access.cpp:175
ERROR:  Error attempting to create data dir: C:/Users/flim/AppData/Roaming/Godot
/app_userdata/my_space_shooter3
At:  core\os\os.cpp:399
attempt set state: (current:4), (sources:res://game.gd, res://player.gd, res://P
rojectileMgr.gd, res://SpawnMgr.gd, res://tomato.gd, res://enemy1path.gd, res://
pathfollow.gd, res://ProjectileNode.gd, res://EnemyNode.gd, res://player_project
ile_A.gd)
UPDATING: res://game.gd
UPDATING: res://player.gd
UPDATING: res://ProjectileMgr.gd
UPDATING: res://SpawnMgr.gd
UPDATING: res://tomato.gd
UPDATING: res://enemy1path.gd
UPDATING: res://pathfollow.gd
UPDATING: res://ProjectileNode.gd
UPDATING: res://EnemyNode.gd
UPDATING: res://player_projectile_A.gd
EXEC PATHP??: E:\Users\flim\Documents\GitHub\godot\bin\godot.windows.tools.exe
running: E:\Users\flim\Documents\GitHub\godot\bin\godot.windows.tools.exe -path
E:/Users/flim/Documents/GitHub/my_projects/space_shooter3 -rdebug localhost:6007
 res://level1.scn
running cmdline: "E:\Users\flim\Documents\GitHub\godot\bin\godot.windows.tools.e
xe" "-path" "E:/Users/flim/Documents/GitHub/my_projects/space_shooter3" "-rdebug
" "localhost:6007" "res://level1.scn"
ERROR: PacketPeerStream::set_stream_peer:  Condition ' p_peer.is_null() ' is tru
e.
At:  core\io\packet_peer.cpp:228
EXEC PATHP??: E:\Users\flim\Documents\GitHub\godot\bin\godot.windows.tools.exe
EXEC PATHP??: E:\Users\flim\Documents\GitHub\godot\bin\godot.windows.tools.exe
opening pack E:\Users\flim\Documents\GitHub\godot\bin\godot.windows.tools.exe, -
1, -1
opening pack E:/Users/flim/Documents/GitHub/my_projects/space_shooter3/data.pck,
 -1, -1
ERROR: DirAccess::make_dir_recursive:  Method/Function Failed, returning: err
At:  core\os\dir_access.cpp:175
ERROR:  Error attempting to create data dir: C:/Users/flim/AppData/Roaming/Godot
/app_userdata/my_space_shooter3
At:  core\os\os.cpp:399
ERROR: Area2D::_clear_monitoring:  Condition ' !node ' is true. Continuing..:
At:  scene\2d\area_2d.cpp:234
ERROR: Area2D::_clear_monitoring:  Condition ' !node ' is true. Continuing..:
At:  scene\2d\area_2d.cpp:234
ERROR: Area2D::_clear_monitoring:  Condition ' !node ' is true. Continuing..:
At:  scene\2d\area_2d.cpp:234
ERROR: Area2D::_clear_monitoring:  Condition ' !node ' is true. Continuing..:
At:  scene\2d\area_2d.cpp:234
ERROR: Area2D::_clear_monitoring:  Condition ' !node ' is true. Continuing..:
At:  scene\2d\area_2d.cpp:234
ERROR: Area2D::_clear_monitoring:  Condition ' !node ' is true. Continuing..:
At:  scene\2d\area_2d.cpp:234
ERROR: PacketPeerStream::set_stream_peer:  Condition ' p_peer.is_null() ' is tru
e.
At:  core\io\packet_peer.cpp:228
```

OS: Windows 8.1 64-bit

**Comment by akien-mga** [comment]
Can you still reproduce this issue in the current master branch?

**Comment by nopeson** [comment]
I can reproduce it too, yes.
Respectively I currently have this issue in my project on the godot 2.0 alpha build (10/30/2015).

**Comment by kakoeimon** [comment]
I can reproduce the problem even in 2.01 in linux.

**Comment by kakoeimon** [comment]
Looks like this is not a bug cause call_deferred solves the problem. For example replace queue_free with call_deferred("queue_free")

**Comment by bojidar-bg** [comment]
@kakoeimon does it still happen if you remove the line after `body.queue_free()`?

**Comment by kakoeimon** [comment]
Sorry for the late response.
The error is this.
ERROR: Area2D::_clear_monitoring:  Condition ' !node ' is true. Continuing..:
At:  scene\2d\area_2d.cpp:234
This error comes when you try to remove an object which contains the area which used to make the collision check. With call_deffered the problem is solved. BUt on the other hand this was supposed to be the job of queue_free.
In the example of the original post, the object with the area2d was the bullet and the error comes only if you try to remove the bullet.

**Comment by MarianoGnu** [comment]
I could not replicate this bug at all, can you provide a test case?

**Comment by kakoeimon** [comment]
I cannot isolate the problem. By the way I get this error in my project for the recent game jam.
https://github.com/kakoeimon/Procedural_Chaos.git
The only Area2D in the project is in the res://instances/enemy_bullet.scn

**Comment by mooseyfaith** [comment]
I can replicate the problem:
bullet.queue_free()
area2d.queue_free()

using bullet.call_defered("queue_free") solves the problem
also calling bullet.queue_free() after area2d.queue_free() solves the problem (since this is basically what the defered call does)

**Comment by henriquelalves** [comment]
I also can replicate this problem in LinuxMint (Godot v2.1 Stable).

**Comment by radishes** [comment]
```
ERROR: Area2D::_clear_monitoring: Condition ' !node ' is true. Continuing..:
   At: scene\2d\area_2d.cpp:359
```
I'm getting this error now when I call queue_free() or call_deferred("queue_free") on instances of my Area2D scene. I started getting this error after I added a RigidBody2D child node and a CollisionShape2D grandchild node to it. Here's what my scene looks like:
![nodetree](https://cloud.githubusercontent.com/assets/1106565/22496755/66776b44-e801-11e6-837d-6aeba3191abd.png)

The errors stop if I manually delete all child nodes before I delete the scene. Here's my workaround:
```

<function where scene is deleted>
	delete_all_children(self)
	call_deferred("queue_free")

func delete_all_children(node):
	"""
	Delete all child nodes of node. Some calls to queue_free() can generate errors if this is not done.
	"""
	for N in node.get_children():
		if N.get_child_count() > 0:
			delete_all_children(N)
		else:
			N.queue_free()
```

**Comment by RandomShaper** [comment]
Is this still happening in 2.1.4?

**Comment by akien-mga** [comment]
First of all thank you for your report and sorry for the delay.

We released Godot 3.0 in January 2018 after 18 months of work, fixing many old issues either directly, or by obsoleting/replacing the features they were referring to.

We still have hundreds of issues whose relevance/reproducibility needs to be checked against the [current stable version](https://godotengine.org/download), and that's where you can help us.
**Could you check if the issue that you described initially is still relevant/reproducible in Godot 3.0 or any newer version, and comment about it here?**

For bug reports, please also make sure that the issue contains detailed steps to reproduce the bug and, if possible, a zipped project that can be used to reproduce it right away. This greatly speeds up debugging and bugfixing tasks for our contributors.

Our Bugsquad will review this issue more in-depth in 15 days, and potentially close it if its relevance could not be confirmed.

Thanks in advance.

*Note: This message is being copy-pasted to many "stale" issues (90+ days without activity). It might happen that it is not meaningful for this specific issue or appears oblivious of the issue's context, if so please comment to notify the Bugsquad about it.*

**Comment by akien-mga** [comment]
As there was no answer to the above request, we're closing this issue as inactive.
If you can still reproduce it, please open a new issue against a currently supported release of Godot.
